itou/utils/resume/forms.py

```diff
@@ -0,0 +1,34 @@
+import django.forms as forms
+import requests
+from django.core.validators import URLValidator
+from django.utils.translation import gettext_lazy
+
+
+class ResumeFormMixin(forms.Form):
+    """
+    Handles resume fields for apply and signup jobseeker forms
+    """
+
+    resume_link = forms.CharField(
+        label=gettext_lazy("Lien vers un CV"),
+        help_text=gettext_lazy("Vous pouvez saisir un lien vers un CV de votre choix (LinkedIn, ...)"),
+        required=False,
+        widget=forms.TextInput(attrs={"placeholder": gettext_lazy("Entrez l'adresse de votre CV")}),
+    )
+
+    def clean_resume_link(self):
+        resume_link = self.cleaned_data["resume_link"]
+
+        if resume_link:
+            # Check if valid (also on model)
+            validator = URLValidator(schemes=["https"])
+            validator(resume_link)
+
+            # Check if exists ?
+            try:
+                requests.head(resume_link, allow_redirects=True)
```

Comment:
Ça existait *out of the box* dans Django en 2011 via `verify_exists` avec un `HEAD` et un `GET` en fallback pour les proxys ou serveur refusant les `HEAD` mais ça a été retiré car considéré comme un vecteur de DOS :

> https://github.com/django/django/commit/1a76dbefdf

Toutes les explications ci-dessous :

- https://www.djangoproject.com/weblog/2011/sep/09/security-releases-issued/
- https://docs.djangoproject.com/en/3.0/releases/security/#september-9-2011-cve-2011-4137

Je suggère de retirer cette vérification d'existence.

Reply:
Oui j'avais vu le `verify_exists` en version 1.x, mais je ne savais pas qu'une des raisons du retrait était un possible DOS. Je vais lire tout ça. Merci